Linker error during deb package generation

# syslog-ng

## Version of syslog-ng
3.15.1
```
output of syslog-ng --version
```

## Platform
Ubuntu 16.04.4 LTS

# Issue

## Failure
During deb package generation on my local machine, I have the following error from linker:
```
gcc -DHAVE_CONFIG_H -I. -I../..  -I../../lib -I../../modules -I./lib -I./modules -Wdate-time -D_FORTIFY_SOURCE=2 -pthread -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I../../lib/eventlog/src -I./lib/eventlog/src   -D_BSD_SOURCE -D__BSD_SOURCE -D__FAVOR_BSD -DHAVE_NET_ETHERNET_H -I/usr/include/dbi -I/usr/include/dbi/dbi  -D_GNU_SOURCE -D_DEFAULT_SOURCE -D_LARGEFILE_SOURCE -D_FILE_OFFSET_BITS=64 -I../../libtest -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros   -O2 -g -pthread -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -c -o lib/tests/lib_tests_test_host_resolve-test_host_resolve.o `test -f 'lib/tests/test_host_resolve.c' || echo '../../'`lib/tests/test_host_resolve.c
/bin/bash ./libtool  --tag=CC   --mode=link gcc -I../../libtest -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros   -O2 -g -pthread -g -O2 -fstack-protector-strong -Wformat -Werror=format-security  -Wl,-Bsymbolic-functions -Wl,-z,relro -o lib/tests/test_host_resolve lib/tests/lib_tests_test_host_resolve-test_host_resolve.o ./libtest/libsyslog-ng-test.a ./lib/libsyslog-ng.la -lrt -lnsl -Wl,--export-dynamic -lgmodule-2.0 -pthread -lgthread-2.0 -pthread -lglib-2.0 ./lib/eventlog/src/libevtlog.la  -lcap -lpcre  -ldl -livykis -lcriterion 
libtool: link: gcc -I../../libtest -Wshadow -fcommon -Wno-stack-protector -Wno-unused-parameter -Wno-variadic-macros -O2 -g -pthread -g -O2 -fstack-protector-strong -Wformat -Werror=format-security -Wl,-Bsymbolic-functions -Wl,-z -Wl,relro -o lib/tests/.libs/test_host_resolve lib/tests/lib_tests_test_host_resolve-test_host_resolve.o -Wl,--export-dynamic -pthread -pthread  ./libtest/libsyslog-ng-test.a ./lib/.libs/libsyslog-ng.so -lrt -lnsl -lgmodule-2.0 -lgthread-2.0 -lglib-2.0 ./lib/eventlog/src/.libs/libevtlog.so -lcap -lpcre -ldl -livykis -lcriterion -pthread -Wl,-rpath -Wl,/usr/lib/syslog-ng
./lib/.libs/libsyslog-ng.so: undefined reference to `secret_storage_wipe'
collect2: error: ld returned 1 exit status
Makefile:10079: recipe for target 'lib/tests/test_host_resolve' failed
make[3]: *** [lib/tests/test_host_resolve] Error 1
Makefile:17952: recipe for target 'all-recursive' failed
make[2]: *** [all-recursive] Error 1
Makefile:7363: recipe for target 'all' failed
make[1]: *** [all] Error 2
make[1]: Leaving directory '/home/norberttakacs/src/syslog-ng/ose-packaging/syslog-ng-3.15.1/debian/build-tree'
dh_auto_build: make -j1 returned exit code 2
debian/rules:189: recipe for target 'build' failed
```

## Steps to reproduce

Execute this script in syslog-ng dir:

```
#!/bin/bash
top_dir=$(pwd)

mkdir -p ./ose-build
cd ./ose-build
.././configure
make dist
mkdir -p ../ose-packaging
cp $(ls -Art | tail -n 1) ../ose-packaging/$(ls -Art | tail -n 1 | sed "s/ng-/ng_/g" | sed "s/.tar.gz//g").orig.tar.gz
cd ../ose-packaging
tar xvf $(ls -Art | tail -n 1)
cd $(ls -td -- */)
mkdir -p debian/build-tree/modules/java-modules/.gradle
echo "systemProp.http.proxyHost=proxy.balabit" | tee --append debian/build-tree/modules/java-modules/.gradle/gradle.properties
echo "systemProp.http.proxyPort=3128" | tee --append debian/build-tree/modules/java-modules/.gradle/gradle.properties
echo "systemProp.https.proxyHost=proxy.balabit" | tee --append debian/build-tree/modules/java-modules/.gradle/gradle.properties
echo "systemProp.https.proxyPort=3128" | tee --append debian/build-tree/modules/java-modules/.gradle/gradle.properties
cp -r ${top_dir}/packaging/debian/ . 
sudo apt-get update
sudo apt-get install -y automake libmongoc-dev libbson-dev librabbitmq-dev dh-exec
debchange --urgency medium --newversion 3.15.1-testing`date "+%Y%m%d%H%M%S"` "test release"
dpkg-buildpackage -us -uc 
tar -cf /generated_deb/debs.tar *.deb
cd $(pwd)
```

--- a/configure.ac
+++ b/configure.ac
@@ -1669,6 +1669,8 @@ else
 fi
 YFLAGS="-d"
 
+TOOL_DEPS_LIBS="${TOOL_DEPS_LIBS} \$(top_builddir)/lib/secret-storage/libsecret-storage.la"
+
 enable_value()
 {
         if test "x$1" = "xyes" ; then

--- a/lib/logproto/tests/Makefile.am
+++ b/lib/logproto/tests/Makefile.am
@@ -23,6 +23,7 @@ lib_logproto_tests_test_findeom_CFLAGS	= \
 	-I${top_srcdir}/libtest
 lib_logproto_tests_test_findeom_LDADD	= \
 	${top_builddir}/lib/libsyslog-ng.la \
-	${top_builddir}/libtest/libsyslog-ng-test.a
+	${top_builddir}/libtest/libsyslog-ng-test.a \
+	$(TEST_LDADD)
 lib_logproto_tests_test_findeom_SOURCES = \
 	lib/logproto/tests/test_findeom.c

--- a/syslog-ng/Makefile.am
+++ b/syslog-ng/Makefile.am
@@ -11,7 +11,9 @@ syslog_ng_syslog_ng_SOURCES	= syslog-ng/main.c
 # libtool, because in mixed mode libtool is not used for the final linking
 # phase.  See the comment in the configure script for details.
 
-syslog_ng_syslog_ng_LDADD		= -L${top_builddir}/lib/.libs -lsyslog-ng @SYSLOGNG_DEPS_LIBS@
+syslog_ng_syslog_ng_LDADD		= -L${top_builddir}/lib/.libs \
+					  -lsyslog-ng @SYSLOGNG_DEPS_LIBS@ \
+					  $(top_builddir)/lib/secret-storage/libsecret-storage.la
 syslog_ng_syslog_ng_LINK		=  @SYSLOGNG_LINK@
 syslog_ng_syslog_ng_DEPENDENCIES	= lib/libsyslog-ng.la
 

